### PR TITLE
Update structure_tensor docstring to include per-axis sigma.

### PR DESCRIPTION
**Pull request by mkcor**
## Description

@grlee77 I also included sanity checks for `sigma`. This contribution addresses https://github.com/scikit-image/scikit-image/pull/5012#discussion_r562884140.

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

**Review by grlee77 — APPROVED** [review]
Thanks, this looks good. 

The test function could potentially be refactored to use `@pytest.mark.parametrize('ndim', [2, 3])` to reduce duplication slightly, but I think it is fine as is.

File: skimage/feature/corner.py [hunk 1]
```diff
@@ -116,6 +119,14 @@ def structure_tensor(image, sigma=1, mode='constant', cval=0, order=None):
         else:
             order = 'rc'
 
+    if isinstance(sigma, tuple):
+        if len(sigma) <= 1:
+            raise TypeError('single sigma must be passed as float or int')
+        else:
+            if len(sigma) != image.ndim:
+                raise ValueError('sigma must have as many elements as image '
+                                 'has axes')
```
Comment (rfezzani) [inline skimage/feature/corner.py:128]: I think that the special case of length 1 iterable is not that special and that we may allow passing array_like values and not only tuples
```suggestion
    if not np.isscalar(sigma):
        sigma = tuple(sigma)
        if len(sigma) != image.ndim:
            raise ValueError('sigma must have as many elements as image '
                             'has axes')
```

Comment (mkcor) [reply skimage/feature/corner.py:128]: Very true, thanks @rfezzani! I'll update the test.

**Comment by mkcor** [comment]
> The test function could potentially be refactored to use `@pytest.mark.parametrize('ndim', [2, 3])` to reduce duplication slightly, but I think it is fine as is.

Done 585386836d73c91fc53664ff6987cb08b6b2f245. Thanks, @grlee77!

@rfezzani please re-review if you get a chance :)

File: skimage/feature/corner.py [hunk 2]
```diff
@@ -58,9 +58,12 @@ def structure_tensor(image, sigma=1, mode='constant', cval=0, order=None):
     ----------
     image : ndarray
         Input image.
-    sigma : float, optional
+    sigma : float or tuple of float, optional
```
Comment (rfezzani) [inline skimage/feature/corner.py:61]:
```suggestion
    sigma : float or array_like, optional
```

File: skimage/feature/corner.py [hunk 3]
```diff
@@ -58,9 +58,12 @@ def structure_tensor(image, sigma=1, mode='constant', cval=0, order=None):
...
         Standard deviation used for the Gaussian kernel, which is used as a
         weighting function for the local summation of squared differences.
+        If sigma is a tuple, its length must be equal to `image.ndim` and each
```
Comment (rfezzani) [inline skimage/feature/corner.py:64]:
```suggestion
        If sigma is an iterable, its length must be equal to `image.ndim` and each
```

**Review by rfezzani — COMMENTED** [review]
Sorry for the delay @mkor :relaxed:. We just need to update the doc and we are ready to go I think :wink:

**Comment by mkcor** [comment]
> Sorry for the delay @mkor relaxed. We just need to update the doc and we are ready to go I think wink

@rfezzani good catch! Thanks -- fixed. I have taken the liberty of editing your suggestion.

**Review by rfezzani — APPROVED** [review]

**Comment by rfezzani** [comment]
Merging! Thank you @mkcor :wink:

**Pull request merged**